experimental/LieAlgebras/src/simple_lie_algebra.jl

```diff
@@ -0,0 +1,266 @@
+#    Generic._matrix(f)[1, r] = d
+#    return f
+#  end
+
+function symbols(L::SimpleLieAlgebra)
+  return L.s
+end
+
+###############################################################################
+#
+#   String I/O
+#
+###############################################################################
+function Base.show(io::IO, ::MIME"text/plain", L::SimpleLieAlgebra)
+  io = pretty(io)
+  println(io, "Simple Lie algebra")
+  println(
+    io, Indent(), "of type $(string(root_type(L)[1])*string(root_type(L)[2]))", Dedent()
+  )
+  println(io, Indent(), "of dimension $(dim(L))", Dedent())
+  print(io, "over ")
+  print(io, Lowercase(), coefficient_ring(L))
+end
+
+function Base.show(io::IO, L::SimpleLieAlgebra)
+  if get(io, :supercompact, false)
+    print(io, "Simple Lie algebra")
+  else
+    io = pretty(io)
+    print(io, "Simple Lie algebra over ", Lowercase())
+    print(IOContext(io, :supercompact => true), coefficient_ring(L))
+  end
+end
+
+# function Base.show(io::IO, V::SimpleLieAlgebra)
+#   print(io, "Simple Lie Algebra over ")
+#   print(IOContext(io, :compact => true), base_ring(V))
+# end
+
+###############################################################################
+#
+#   Arithmetic operations
+#
+###############################################################################
+
+# Binary operations
+
+function bracket(
+  x::SimpleLieAlgebraElem{C}, y::SimpleLieAlgebraElem{C}
+) where {C<:RingElement}
+  check_parent(x, y)
+  L = parent(x)
+  mat = sum(
+    cxi * cyj * L.struct_consts[i, j] for (i, cxi) in enumerate(coefficients(x)),
+    (j, cyj) in enumerate(coefficients(y))
+  )
+  return L(mat)
+end
+
+###############################################################################
+#
+#   Comparison
+#
+###############################################################################
+
+function ==(L::SimpleLieAlgebra{T}, M::SimpleLieAlgebra{T}) where {T<:RingElement}
+  return base_ring(L) == base_ring(M) &&
+         dim(L) == dim(M) &&
+         root_system(L) == root_system(M)
+end
+
+function Base.hash(x::SimpleLieAlgebra{C}, h::UInt) where {C<:RingElement}
+  b = 0x9072419641dbc7d0 % UInt
+  h = hash(base_ring(L), h)
+  h = hash(dim(L), h)
+  h = hash(mat_space(L), h)
+  h = hash(root_system(L), h)
+  h = hash(root_type(L), h)
+  return xor(h, b)
+end
+
+###############################################################################
+#
+#   Constructors
+#
+###############################################################################
+
+@doc raw"""
+    lie_algebra(R::Ring, S::String, cached::Bool=true) -> SimpleLieAlgebra{elem_type(R)}
+
+Construct the simple Lie algebra over the ring `R` with root system of type `S`
+The internally used basis of this Lie algebra is the Chevalley basis.
+"""
+function lie_algebra(R::Ring, S::Symbol, n::Int64; cached::Bool=true)
+  @req S in [:A, :B, :C, :D, :E, :F, :G] "Unknown Dynkin type"
+  T = elem_type(R)
+  return SimpleLieAlgebra{T}(R, S, n; cached)
```

Comment:
```suggestion
  return SimpleLieAlgebra{elem_type(R)}(R, S, n; cached)
```
You do not need input sanitation here, since you call `root_system` anyway which then checks `S` and `n` for validity.